### PR TITLE
Custom deploy commands for any cloud with s3 compatible object store.

### PR DESCRIPTION
**Pull request by dwhitena**
# What this PR does:

Implements a `custom` deploy command option such that users can deploy pachyderm with k8s in any cloud, while still backing Pachyderm by any S3 compatible object store (like Minio).

# How to test this PR:

To deploy to Google cloud + a Minio (or other s3 compatible object store) instance, follow the [normal deploy](http://docs.pachyderm.io/en/latest/deployment/deploying_on_the_cloud.html#google-cloud-platform) for google minus setting up the bucket.  Then you can deploy via:

```
pachctl deploy custom --persistent-disk google --object-store s3 <persistent disk name> <persistent disk size> <object store bucket> <object store id> <object store secret> <object store endpoint>
```

To deploy to AWS cloud + a Minio (or other s3 compatible object store) instance:

```
pachctl deploy custom --persistent-disk aws --object-store s3 <persistent disk name> <persistent disk size> <object store bucket> <object store id> <object store secret> <object store endpoint>
```

To deploy to Azure cloud + a Minio (or other s3 compatible object store) instance:

```
pachctl deploy custom --persistent-disk azure --object-store s3 <persistent disk uri> <persistent disk size> <object store bucket> <object store id> <object store secret> <object store endpoint>
```

### CODE REVIEW

**Comment by dwhitena** [comment]
I have tested this with Google + Minio.  Need to test with AWS + Minio and Azure + Minio.

**Comment by dwhitena** [comment]
Tested with AWS + Minio.  Just need Azure now.

**Comment by dwhitena** [comment]
After a bit of pain, this is now confirmed working with Azure.  Ready for review!

File: etc/microsoft/create-blank-vhd/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM golang:1.7.1-alpine
+FROM golang:alpine
```
Comment (sjezewski) [inline etc/microsoft/create-blank-vhd/Dockerfile]: Looks like we upgraded to a newer go image. Let's make sure we keep referencing the image by a version tag, even if its the latest tag.

Comment (jdoliner) [inline etc/microsoft/create-blank-vhd/Dockerfile]: I was about to comment on this as well.

**Comment by jdoliner** [comment]
One small comment on the go docker image we're using. Then lgtm.

**Comment by dwhitena** [comment]
Ok @jdoliner and @sjezewski.  I added the specific tag.  Thanks!

**Comment by jdoliner** [comment]
Cool, feel free to hit the big green button as soon as CI passes.

**Pull request merged**